### DISCUSSION TITLE
Revshare on Rafiki

### DISCUSSION THREAD

**Discussion by cairin** [interledger/rafiki#40] [Q&A]
From @ProgNovel:

My greatest concern about Rafiki payments is that how we do revshare on it. With Web Monetization we have probability rev share which is a decent solution, but that wouldn't work smoothly on Rafiki especially on tippings and subscriptions. 

I wrote a simple thought on it on https://community.webmonetization.org/radhyr/dividend-revenue-sharing-12ig, but will appreciate if others can enlighten me with a more sophisticated way of doing revshare on Rafiki.

PS: I happened to create a discussion of revenue share related discussions thread on Rafiki, please hop on if you're interested:
https://community.webmonetization.org/radhyr/revenue-sharing-on-rafiki-mega-thread-2565

**Comment by totorogendut** [comment]
Thanks for the repost. Hope people will put more attention to this issue since we previously have a pretty decent revshare option for Web Monetization API, but not native Interledger payments yet.

**Comment by matdehaast** [comment]
@ProgNovel I am going to caveat my answer with this is just a thought experiment and not what will be settled on.

Much of the model we are proposing with Rafiki and its use of Open Payments is delegated access to third parties. For revenue sharing the way I would envision it would work is that when a User authorizes a set amount for a platform to be able to send payments. Those payments can be split however the platform sees fit. Thus they would be able to send partial amounts to the intended recipient as well as themselves.

A more concrete example would be if I have a video streaming platform and want to enable tipping to content creators but also want to take a cut of the revenue of 20%. When a user Tips the creator, say $1, I as the platform will make a payment for $0.80 to the content creators account and then a $0.20 payment to my own account.

**Reply by totorogendut** [reply]
Sorry for the late reply and thanks for the respond. It'd be great if there any open source solution for it since smaller orgs that don't have resource for creating a decent sized platforms might be interested in it.
